v1/gitlab/test-binary-fields-selective_code_owner_removals.rego

```diff
@@ -0,0 +1,59 @@
+package verify
+
+import future.keywords.in
+
+default allow := false
+default violations := []
+
+default selective_code_owner_removals := false
+
+selective_code_owner_removals := input.config.args.selective_code_owner_removals 
+
+verify = v {
+    v := {
+        "allow": allow,
+        "violation": {
+            "type": "Verify that the binary field: 'selective_code_owner_removals' is set correctly",
+            "details": violations,
+        },
+        "summary": [{
+            "allow": allow,
+            "reason": reason,
+            "violations": count(violations),
+        }],
+    }
+}
+
+allow {
+    count(violations) == 0
+}
+
+reason = v {
+    allow
+    v := "All binary fields: 'selective_code_owner_removals' are properly set "
+}
+
+reason = v {
+    not allow
+    v := "At least one of the binary fields: 'selective_code_owner_removals' are not properly set"
+}
+
+
+
+# Violation has been switched here as a list instead of a set
+violations := [r |
+
+    project := object.remove(input.evidence.predicate.content, {"metadata"})[_]
+
+    # r := { "s": project.project.result_object.approval_settings.selective_code_owner_removals}
+    selective_code_owner_removals != project.project.result_object.approval_settings.selective_code_owner_removals
```

Comment:
Since you have only one check here and don't iterate through any objects, better move this to `allow` field to have a better focus and leave `violations` for additional details. Take a look at `v1/images/fresh-image.rego`, it's a good example of such check.